wallet-new/integration/Types.hs

```diff
@@ -39,19 +34,14 @@ type WalletTestMode m =
     )
 
 -- | The probability type that captures the chance of
--- our random action selection choosing a specific @Action@.
--- The number is restricted to 1 - 100 - see @createProbability@.
-data Probability = Probability { getProbability :: Int }
+-- our random action selection choosing a specific 'Action'.
+--
+-- An 'Action' is paired with a 'Weight'. All of the 'Weight's are summed
+-- up, and the eventual probability is the weight divided by the total
+-- weight.
+newtype Weight = Weight { getWeight :: Int }
```

Comment:
I switched from a `0-100` probability to a weight model. This drops all of the partiality involved and makes it simpler to work with.